src/bundles/peer-locations.js

```diff
@@ -300,44 +148,65 @@ class PeerLocationResolver {
       ...opts.cache
     })
 
+    this.peersToAddrs = {}
```

Comment:
Mapping `peersToAddrs` seems to short circuit the ipv4 lookup cache, and means that a peerId once seen will always be mapped to the same ip address, even if it tells us it's ip address has changed.

Reply:
Fixed!